akka-stream/src/main/scala/akka/stream/impl/BoundedSourceQueue.scala

```diff
@@ -133,6 +143,12 @@ import akka.stream.stage.{ GraphStageLogic, GraphStageWithMaterializedValue, Out
         if (setDone(Done(QueueOfferResult.Failure(ex))))
           Logic.callback.invoke(()) // if this thread won the completion race also schedule an async callback
       }
+
+      override def whenReady(): Future[akka.Done] = {
+        val promise = Promise[akka.Done]()
+        awaitingSubscribers.offer(promise)
```

Comment:
The only potential problem I can see here - that a single thread can call `whenReady` multiple times and which makes not much sense. I'm not yet sure how whether we need to try to handle that. Could you please share your opinion.